### ISSUE TITLE
Analysis of Hippocampus-CA1 model with BBP-tests

### ISSUE THREAD

**Issue by alex4200** [cnr-ibf-pa/hbp-bsp-issues#332]
## Summary
  
Aspect | Detail
------------ | -------------
Summary | Notebooks with existing BBP-tests for CA1, that will be registered in the HBP-VF. Tests will be able to register the results.
Group |
State |
Expert |
User |
Deadline | To be defined. It depends on setup completion to access HPC resources, e.g. BluePy installation on JURECA, access to CA1-model release, submission of jobs through UNICORE, etc
Target audience |
Target interface | Notebooks
HPC Requirements |
Dependencies |
Nominal runtime |
  
**Remove anything that does not apply**
## List of features
*What does your use case do (or the new/changed features)? What activities can users perform while using it? What makes it different from similar use cases? List the main functions that you will build into your product here. Also specify the priority ‘mandatory’, ‘important, ‘nice to have’.*
  
## Acceptance Tests
*Define here the acceptance tests to evaluate the use case’s compliance with the requirements as defined above. Also possible end users for testing can be included here.*
  
### System
*Does this use case require any special system requirements (any special software package)? Does this use case require HPC access? If so, which ones? List detailed system requirements here.*
  
### Performance
*What are additional performance requirements for this use case? What time should it take to run from start to end?*

**Comment by pedroernesto** [comment]
Better to move the deadline to June/2019.  It depends on setup completion of several tools and HPC resources

**Comment by pedroernesto** [comment]
Most of BBP-tests currently do not include any score to register on the VF, but plots and visual comparisons. A lot of coding is needed yet, and getting familiar with their own approaches and tools as well. More importantly, the right statistical test to produce a relevant score for each particular situation must be found.
 
I propose to eliminate this issue. I can then focus on more realistic goals related with issues #323 and #331

**Comment by apdavison** [comment]
I don't think we should eliminate this, it is very central to our work. I think the milestone of M18 seems reasonable.

**Comment by pedroernesto** [comment]
Ok. 
Actually, one of those BBP cases with figures and no scores, is the one considered in issue #331: a real test for synapses distributions in the CA1 circuit model is missing. So, it would be clearer if the content of issue #331 were actually one of the notebooks I mentioned in the title of this issue #332. Issue #331 is redundant.

**Comment by pedroernesto** [comment]
@alex4200 , I have spoken to Andrew about this use-case and we agreed that to start we will need:
 
(1) to have saved a model version of the CA1 circuit in some repository accessible from the Collab (currently > 300 Gb, but can likely be reduced, since not all files are needed for the use-case)

(2) that the user have access to BBP-VPN from the Collab, to install a few BBP software (e.g. BluePy). This will happen anytime a user runs the use-case for the first time.

**Comment by alex4200** [comment]
Moved to later milestone

**Comment by alex4200** [comment]
@pedroernesto Where is that CA1 circuit saved currently? Can you give me the location?

**Comment by alex4200** [comment]
@pedroernesto Before we can proceed, we need to clarify some issues, since the model you want to use is in the realm of BBP, and the Collaboratory and use cases on the Brain Simulation Platform in the realm of HBP. 
I will keep you posted.

**Comment by romani79** [comment]
@pedroernesto, putting the model in BSP means sharing the model with everyone. We have to discuss which versions can be shared and in which modalities. We will have a meeting in 10 days and I hope we will have a plan.

**Comment by pedroernesto** [comment]
I understand your point, no problem. 
Specifically, I told @alex4200 about the following directories of (a little bit old) release of the model. But maybe @romani79  can reduce the list of dirs/files needed, even more. In fact, this usecase will be based on a previous version of some his notebooks for circuit analysis. The directories are part of proj_dir = '/gpfs/bbp.cscs.ch/project/proj42/':

$proj_dir/circuits/CA1.O1/20181114 (excluding file SEGMENT_payload.dat of this directory)
$proj_dir/entities/morphologies/20180417/
$proj_dir/entities/emodels/20180504b/
$proj_dir/entities/bionames/20181114
$proj_dir/entities/dev/atlas/O1-243

The tests are about circuit structure only,  no simulation results.

**Comment by alex4200** [comment]
@romani79 Do you have any updates regarding this issue? In your last comment you mentioned a meeting - do you have any news from that?

**Comment by romani79** [comment]
We are going to release a rat CA1 microcircuit, but not the full-scale.
The release date should be defined. It could be at the same time of the release of the MOOC or earlier.

**Comment by pedroernesto** [comment]
@romani79 Was this mircro-circuit already released in some computer?

**Comment by romani79** [comment]
@pedroernesto, the O1 is available in a private HBP storage and it will be released once the paper is out. You can access it as part of the hippocampus team, but this is still confidential.
If you would like to access it, please ask @antonelepfl.

**Comment by pedroernesto** [comment]
Ok, the point was granting access to that circuit to have this use-case on the Collab. But since not everybody on the Collab belongs to HBP, I guess this issue will stay opened till the paper gets published.

**Comment by alex4200** [comment]
@romani79 @pedroernesto  Should this task be moved to SGA3?

**Comment by pedroernesto** [comment]
As Armando says above, the circuit will not be released till the publication of the paper. But the circuit might still need fixes and the simulations more analysis and validations. Besides, the reviewing process after submission. This might take about 2 years I think. So, I propose to close this issue temporarily and then re-open it again afterwards.
